### ISSUE TITLE
ImageViewSet.data redirect error for S3 files

### ISSUE THREAD

**Issue by banesullivan** [ResonantGeoData/ResonantGeoData#642]
If an Image (it's ChecksumFile) is stored remotely on S3, this endpoint errors:

https://github.com/ResonantGeoData/ResonantGeoData/blob/efb2ab026c68ffa852a399aa122d9b38c320afcc/django-rgd-imagery/rgd_imagery/rest/viewsets.py#L67-L71

We should provide a workaround of some sort as the Python client calls this endpoint expecting data to be returned.

Perhaps for S3 files, we can just pass back the S3 URL and let the user figure it out?

Otherwise, we'd have to proxy the whole file through RGD's web server

**Comment by mcovalt** [comment]
We should probably just send the `s3://` protocol URL as text.

> Otherwise, we'd have to proxy the whole file through RGD's web server

Django is not really equipped to do this. This would exhaust our available threads if deployed without gevent or eventlet. Even then, I'm not sure how capable those libraries are suited for holding long TCP sessions.

An alternative is much more complex: redirect to presigned URLs of the data. This involves issuing commands with the S3 provider's API to acquire the resigned URLs.

**Comment by banesullivan** [comment]
> We should probably just send the s3:// protocol URI as text.

Let's do this. @mcovalt, could you implement when you have a chance (lowest priority in your queue)

**Comment by mvandenburgh** [comment]
Do we know *why* the endpoint errors? To my knowledge (emphasis on **my**, I could be wrong here), since `ChecksumFile.file` is a `S3FileField`, calling `checksum_file.file.url` should return a signed download URL, which should work with `HttpResponseRedirect` without any problems.

**Comment by banesullivan** [comment]
@mvandenburgh, you're right when working with S3FF but the `ChecksumFile` model has two modes: S3FF or a URL string, see

https://github.com/ResonantGeoData/ResonantGeoData/blob/71c4cd38fa5ec830a2b62765f1dd3772e22debfa/django-rgd/rgd/models/file.py#L38-L40

https://github.com/ResonantGeoData/ResonantGeoData/blob/71c4cd38fa5ec830a2b62765f1dd3772e22debfa/django-rgd/rgd/models/file.py#L73-L75

https://github.com/ResonantGeoData/ResonantGeoData/blob/71c4cd38fa5ec830a2b62765f1dd3772e22debfa/django-rgd/rgd/models/file.py#L83-L94

This issue is referring to the case of `FileSourceType.URL` `ChecksumFile` records that have an `s3://` URL.

For instance:

```ipython
In [1]: f = ChecksumFile(type=2, url='s3://sentinel-cogs/sentinel-s2-l2a-cogs/2020/S2A_34JCL_20200309_0_L2A/S2A_34JCL_20200309_0_L2A.json')

In [2]: f.get_url()
Out[2]: 's3://sentinel-cogs/sentinel-s2-l2a-cogs/2020/S2A_34JCL_20200309_0_L2A/S2A_34JCL_20200309_0_L2A.json'
```

So when we do:

https://github.com/ResonantGeoData/ResonantGeoData/blob/efb2ab026c68ffa852a399aa122d9b38c320afcc/django-rgd-imagery/rgd_imagery/rest/viewsets.py#L67-L71

We are trying to redirect to an `s3://` URL which does not work.

Instead, what we should do remake the endpoint to redirect only for `ChecksumFile` records of `type = FileSourceType.FILE_FIELD` and have the `FileSourceType.URL` type return a text record of the URL (not signed)

**Comment by mvandenburgh** [comment]
Ohh I see, I wasn't thinking about URL files, thanks for the explanation. I'll go ahead and tweak the endpoint to return the URL as a string for `FileSourceType.URL` files

**Comment by banesullivan** [comment]
It'd be good to make sure this is handled in 

https://github.com/ResonantGeoData/ResonantGeoData/blob/a0e5d346a72d1926d468491280546639e1cd02b9/django-rgd/rgd/rest/viewsets.py#L62-L64

and have

https://github.com/ResonantGeoData/ResonantGeoData/blob/efb2ab026c68ffa852a399aa122d9b38c320afcc/django-rgd-imagery/rgd_imagery/rest/viewsets.py#L67-L71

call that
